### PR TITLE
repo_utils: improve the handling about another git ssh expression

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
add some improvement about convert repo url to dirname

Signed-off-by: Vicente Cheng <freeze.bilsted@gmail.com>

### CODE REVIEW

**Comment by kshtsk** [comment]
retest this please

**Comment by kshtsk** [comment]
ok to test

**Comment by kshtsk** [comment]
jenkins test

File: teuthology/repo_utils.py [hunk 1]
```diff
@@ -281,13 +281,17 @@ def url_to_dirname(url):
...
         https://github.com/ceph/ceph -> github.com_ceph_ceph
         https://github.com/liewegas/ceph.git -> github.com_liewegas_ceph
         file:///my/dir/has/ceph.git -> my_dir_has_ceph
     """
     # Strip protocol from left-hand side
-    string = re.match('.*://(.*)', url).groups()[0]
+    if "@" in url:
+        string = re.match('.*@(.*)', url).groups()[0]
+    else:
+        string = re.match('.*://(.*)', url).groups()[0]
```
Comment (kshtsk) [inline teuthology/repo_utils.py]: I think it is better to use single re for this: `(?:.*://|.*@)(.*)`

**Comment by kshtsk** [comment]
ok to test

**Comment by Vicente-Cheng** [comment]
@kshtsk sorry for late replay...

I think you are correct with single re. fixed that.
please check again.

thanks

**Comment by kshtsk** [comment]
retest this please

**Comment by kshtsk** [comment]
@susebot run deploy

**Comment by toabctl** [comment]
Having a unittest for that would be nice imo

**Comment by Vicente-Cheng** [comment]
OK, I would add the related unittest.

**Comment by Vicente-Cheng** [comment]
@kshtsk @toabctl ,
unittest added, please check again.
thanks

**Comment by Vicente-Cheng** [comment]
retest this please

**Pull request merged**